### ISSUE TITLE
speed up swupd-server testing

### ISSUE THREAD

**Issue by pohly** [clearlinux/swupd-server#73]
I'm currently adding automated testing of meta-swupd. One bottleneck is the slow compression of pack-os-core-from-0.tar with "bsdtar -J". It's single-CPU and fully occupies one CPU for a minute or more.

Given that the file is never actually getting published and (if at all) only used inside the same machine, would it make sense to make the compression level configurable, not just for this file but everywhere? Then unit tests that just do basic sanity testing and don't care about compression rates can be sped up by using a faster compression or even no compression at all.

**Comment by bryteise** [comment]
I know there was some push to be able to skip the compression step for testing. I'm not sure what has become of that though.

**Comment by phmccarty** [comment]
@bryteise Yes, there was some discussion recently about skipping compression steps entirely. I would also be in favor of making compression levels configurable, but the fact that swupd-server will try gzip/bzip2/xz and choose between the three, configurable levels for all supported compression types would be needed.

**Comment by pohly** [comment]
Depending on the test, creating the pack-os-core-from-0.tar can be skipped entirely because it is only needed for installing from scratch. I'll add that to meta-swupd.

But in general the feature request still stands.
